GitUI/GitModuleControl.cs

```diff
@@ -69,9 +69,11 @@ public IGitUICommandsSource UICommandsSource
             }
         }
 
-        /// <summary>Gets the <see cref="UICommandsSource"/>'s <see cref="GitUICommands"/> reference.</summary>
+        /// <summary>
+        ///  Gets the <see cref="UICommandsSource"/>'s <see cref="GitUICommands"/> reference.
+        /// </summary>
         [Browsable(false)]
-        public GitUICommands UICommands => UICommandsSource.UICommands;
+        public override GitUICommands UICommands => UICommandsSource.UICommands;
```

Comment:
This change concerns me, feels like we're now mixing unrelated behaviours here.

What was the problem with the TryGetUiCommands API?

Reply:
> What was the problem with the TryGetUiCommands API?

1. `TryGetUICommands` threw exceptions instead of returning `false`!
2. There is no situation where the ServiceProvider is optional. (Calls depending on this shall be deferred to `OnRuntimeLoad`.)
    `LoadHotkeys` is the only user of `TryGetUICommands`.
    (`Visible` is not a the correct replacement of `_loaded` because `OnRuntimeLoad` is called once when a control is shown for the first time. Later it could be invisible because of switching tabs. Reloading hotkeys in every `OnShow` would not be good.)

This is not mixing unrelated things.
For GitModuleControls, the former `TryGetUICommands` / new `virtual` property `UICommands` is not called but the override in GitModuleControl, which directly returns the instance without the cast.
That's why the implementation can be shortened.
(I kept the full variant for having a smaller changeset in the review, and for avoiding the discussion whether there will be controls which implement `IGitModuleControl` but do not inherit from `GitModuleControl`.)
```
        /// <summary>
        ///  Attempts to find an instance of <see cref="IGitUICommands"/>.
        /// </summary>
        /// <remark>
        ///  The instance of <see cref="IGitUICommands"/>
        ///  either directly assigned to the control (if the control inherits from <see cref="GitModuleControl"/> which overrides this property)
        ///  or to the parent form (if the form implements <see cref="IGitModuleForm"/>).
        /// </remark>>
        /// <returns>
        ///  <see langword="true"/>, if an instance of <see cref="IGitUICommands"/> is found; <see langword="false"/>, otherwise.
        /// </returns>
        /// <exception cref="InvalidOperationException">
        ///  If this control is not a <see cref="GitModuleControl"/>) and is not placed in a <see cref="IGitModuleForm"/>.
        /// </exception>
        public virtual IGitUICommands UICommands
            => FindForm() is IGitModuleForm form ? form.UICommands
                : throw new InvalidOperationException($"no chance to get {nameof(UICommands)}");
```

Reply:
I have thought about it again.

> feels like we're now mixing unrelated behaviours here.

Yes, you are right.

> What was the problem with the TryGetUiCommands API?

In addition to the above points, also the second part of the identifier and the return type - in the very first place.
Because, if a control needs `UICommands` it shall implement `IGitModuleControl` / inherit from `GitModuleControl`, but not try to downcast to `IGitModuleControl`.
We just need the `ServiceProvider` for `GitExtensionsControl`.
(With this, a better name for `_loaded` became obvious. `_hotkeysLoaded` would not have matched to exact meaning.)